components/engine-camel/src/main/java/org/eclipse/dirigible/components/engine/camel/processor/CamelProcessor.java

```diff
@@ -23,16 +24,14 @@
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
-import java.nio.charset.StandardCharsets;
+import java.io.IOException;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 
 @Component
 @Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
```

Comment:
I was thinking about this and the possibility for one of our synchronisers to call a camel route when a js file is saved via the js api. If the processor instance, which wraps the SpringBootCamelContext and is obtained via the SpringBeanProvider in js does not have the specific route loaded it will fail. That's why i thought having a single processor instance for all routes was simpler, but i'm not really sure about this since it involves synchronisation of camel and js files and js api calls which get the processor instance

Reply:
How does having a single instance help in this case? Are you talking about concurrency problems?

Reply:
Ok, I saw now there is internal state in this class that is probably safer to be in some singleton. It’s probably a good idea to separate this into another class - CamelInstancesStore or something similar but no need to do that in this PR.

Reply:
Seems to work as without it, removed it